src/flows/shared/steps/OrganizeCollection/Sidebar/Sidebar.tsx

```diff
@@ -196,17 +192,62 @@ const SidebarTokens = ({ nftFragmentsKeyedByID, tokens }: SidebarTokensProps) =>
     mount();
   }, [nftFragmentsKeyedByID, reportError, tokens]);
 
-  return (
-    <>
-      {displayedTokens.map((editModeToken) => (
+  /**
+   * We render a row with three token.
+   */
+  const rowRenderer = ({ key, style, index }: ListRowProps) => {
+    const items = [];
+
+    // add blank block button at the beginning of list
+    if (index === 0) {
+      items.push(
+        <StyledAddBlankBlock onClick={handleAddBlankBlockClick}>
+          <StyledAddBlankBlockText>Add Blank Space</StyledAddBlankBlockText>
+        </StyledAddBlankBlock>
+      );
+    }
+
+    const fromIndex = index === 0 ? 0 : index * COLUMN_COUNT - 1;
+
+    // If first row, get two tokens (since we already have the add blank block button)
+    // If second row onwards, get three tokens
+    const lastAddedTokenIndex = fromIndex + (index === 0 ? COLUMN_COUNT - 1 : COLUMN_COUNT);
+
+    const toIndex = Math.min(lastAddedTokenIndex, displayedTokens.length);
+
+    for (let i = fromIndex; i < toIndex; i++) {
+      const editModeToken = displayedTokens[i];
+      items.push(
         <SidebarNftIcon
           key={editModeToken.token.id}
           tokenRef={nftFragmentsKeyedByID[editModeToken.token.id]}
           editModeToken={editModeToken.token}
           previewUrlSet={editModeToken.previewUrlSet}
         />
-      ))}
-    </>
+      );
+    }
+
+    return (
+      <Selection key={key} style={style}>
+        {items}
+      </Selection>
+    );
+  };
+
+  return (
+    <StyledListTokenContainer>
+      <AutoSizer>
```

Comment:
Didn't implement any `CellMeasurer` since we already have a fixed height and width of the child which is 60 x 60 - `SidebarNftIcon`